### PR TITLE
Introduce `add` family of commands for company IAM

### PR DESCRIPTION
**Pull request by JGiola**
## What this PR is for?

This PR will introduce the new `add` family of command for `company iam`.

Those command are for:

- add a new user in a company with a given role
- create a new group in a company with a given role
- create a service account in a company with a given role
- add one or more user to an existing group in a company

### CODE REVIEW

File: internal/cmd/company/iam/group/create.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			client, err := client.APIClientForConfig(restConfig)
+			cobra.CheckErr(err)
+
+			err = createNewGroup(cmd.Context(), client, restConfig.CompanyID, args[0], resources.ServiceAccountRole(options.IAMRole))
+			cobra.CheckErr(err)
+		},
+	}
+
+	options.CreateNewGroupFlags(cmd.Flags())
+	err := cmd.RegisterFlagCompletionFunc("role", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```
Comment (davidebianchi) [inline internal/cmd/company/iam/group/create.go:51]: Those roles should be dynamic? There is a role api, which could be used to dynamically configure this completion. What do you think?
In the future, probably the roles will change

File: internal/cmd/company/iam/group/create.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	if err != nil {
+		// we panic here because if we reach here, something nasty is happening in flag autocomplete registration
+		panic(err)
+	}
+
+	return cmd
+}
+
+func createNewGroup(ctx context.Context, client *client.APIClient, companyID, groupName string, role resources.ServiceAccountRole) error {
+	if !resources.IsValidServiceAccountRole(role) {
```
Comment (davidebianchi) [inline internal/cmd/company/iam/group/create.go:71]: Also this check is static

File: internal/cmd/company/iam/user/create.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+			client, err := client.APIClientForConfig(restConfig)
+			cobra.CheckErr(err)
+
+			err = addUserToCompany(cmd.Context(), client, restConfig.CompanyID, options.UserEmail, resources.ServiceAccountRole(options.IAMRole))
+			cobra.CheckErr(err)
+		},
+	}
+
+	options.AddNewUserFlags(cmd.Flags())
+	err := cmd.RegisterFlagCompletionFunc("role", func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```
Comment (davidebianchi) [inline internal/cmd/company/iam/user/create.go:51]: Also here, static completion

**Review by davidebianchi — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,13 +7,24 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- the company iam list commands now return the id of the entities as well
 - update go version to 1.21.5
 - update logr to v1.4.1
 - update uuid to 1.5.0
 - update exp to v0.0.0-20231219180239-dc181d75b848
 
+### BREAKING
+
+- move serviceaccount commads under `company iam add`
+
```
[the remark this answers is not in the document]
Comment (JGiola) [reply CHANGELOG.md]: done

File: docs/30_commands.md [hunk 5]
```diff
@@ -210,15 +223,17 @@ Available flags for the command:
 - `--insecure-skip-tls-verify`, to disallow the check the validity of the certificate of the remote endpoint
 - `--context`, to specify a different context from the currently selected one
 - `--company-id`, to set the ID of the desired Company
+- `--output`, optional flag to save the service account json description in a file at the provided path
```
[the remark this answers is not in the document]
Comment (JGiola) [reply docs/30_commands.md]: I've changed the docs to better reflect what we are saving

**Review by davidebianchi — APPROVED** [review]
LGTM

**Pull request merged**